Apps/Database/src/Delegates/DbProfileDelegate.cs

```diff
@@ -251,20 +251,42 @@ public DbResult<List<UserProfile>> GetClosedProfiles(DateTime filterDateTime, in
         }
 
         /// <inheritdoc/>
-        public IDictionary<DateTime, int> GetDailyLoggedInUsersCount(TimeSpan offset)
+        public IDictionary<DateTime, int> GetLoggedInUsersCount(DateTimeOffset startDateTimeOffset, DateTimeOffset endDateTimeOffset)
         {
-            Dictionary<DateTime, int> dateCount = this.dbContext.UserProfile
-                .Select(x => new { x.HdId, x.LastLoginDateTime })
-                .Concat(
-                    this.dbContext.UserProfileHistory.Select(x => new { x.HdId, x.LastLoginDateTime }))
-                .Select(x => new { x.HdId, lastLoginDate = GatewayDbContext.DateTrunc("days", x.LastLoginDateTime.AddMinutes(offset.TotalMinutes)) })
-                .Distinct()
-                .GroupBy(x => x.lastLoginDate)
-                .Select(x => new { lastLoginDate = x.Key, count = x.Count() })
-                .OrderBy(x => x.lastLoginDate)
-                .ToDictionary(x => x.lastLoginDate, x => x.count);
+            TimeSpan offset = startDateTimeOffset.Offset;
+            var userProfileQuery = this.dbContext.UserProfile
+                .Where(u => u.LastLoginDateTime >= startDateTimeOffset.UtcDateTime && u.LastLoginDateTime <= endDateTimeOffset.UtcDateTime)
+                .Select(u => new
+                {
+                    u.HdId,
+                    u.LastLoginDateTime,
+                });
 
-            return dateCount;
+            var userProfileHistoryQuery = this.dbContext.UserProfileHistory
+                .Where(u0 => u0.LastLoginDateTime >= startDateTimeOffset.UtcDateTime && u0.LastLoginDateTime <= endDateTimeOffset.UtcDateTime)
+                .Select(u0 => new
+                {
+                    u0.HdId,
+                    u0.LastLoginDateTime,
+                });
```

Comment:
Nit: the `u0` parameters can be swapped to `u` without conflicting with the parameters in the other query.

Reply:
Yeah, it was an evolution but I've updated.